packages/cli/src/specs/__snapshots__/repo-builder.spec.ts.snap

```diff
@@ -1,5 +1,66 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`Repo Builder all in a class: all in a class 1`] = `
+"import { Store, createState, withProps } from '@ngneat/elf';
+import { withEntities, selectAll, deleteEntities, addEntities, updateEntities, withUIEntities, withActiveId, selectActiveEntity, setActiveId, withActiveIds, selectActiveEntities, toggleActiveIds } from '@ngneat/elf-entities';
+import { Observable } from 'rxjs';
+import { withRequestsCache, withRequestsStatus } from '@ngneat/elf-requests';
+
+export interface TodoUI {
+  id: number;
+}
+
+export interface Todo {
+  id: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface TodosProps {
+}
+
+export class TodosRepository {
+  activeTodos$: Observable<Todo[]>;
+  activeTodo$: Observable<Todo | undefined>;
+  todos$: Observable<Todo[]>;
+
+  private store: ReturnType<TodosRepository['createStore']>;
+
+  constructor() {
+    this.store = this.createStore();
+    this.todos$ = this.store.pipe(selectAll());
+    this.activeTodo$ = this.store.pipe(selectActiveEntity());
+    this.activeTodos$ = this.store.pipe(selectActiveEntities());
+  }
+
+  deleteTodo(id: Todo['id']) {
+    this.store.update(deleteEntities(id));
+  }
+
+  addTodo(todo: Todo) {
+    this.store.update(addEntities(todo));
+  }
+
+  updateTodo(id: Todo['id'], todo: Partial<Todo>) {
+    this.store.update(updateEntities(id, todo));
+  }
+
+  setActiveId(id: Todo['id']) {
+    this.store.update(setActiveId(id));
+  }
+
+  toggleActiveIds(ids: Array<Todo['id']>) {
+    this.store.update(toggleActiveIds(ids));
+  }
+
+  private createStore(): Store<{ name: string; state: typeof state; config: typeof config; }> {
```

Comment:
I'm not sure what is `typeof state` and `typeof config`

Reply:
It infers a type of the `state` and `config` that are created inside this function.

Reply:
Are you sure? How it make sense?

Reply:
I don't know when it was implemented in TS but it works 
<img width="595" alt="image" src="https://user-images.githubusercontent.com/31760529/154855997-f6a5690c-8372-41c3-a525-9f44bdaf5835.png">

Reply:
<img width="1204" alt="image" src="https://user-images.githubusercontent.com/31760529/154856042-5a9fc61f-d738-4411-a69b-19f8b0205d00.png">

Reply:
Wow. Did not know that we can do this

Reply:
Same thing, just discover it while making a pr😄
https://www.typescriptlang.org/play?#code/FAYwNghgzlAEAqBTKAXWBvYtuwPqoHsAnRALlgCVEUBXIgO3gE8AHRAHiVQG0ByEEhBSIAyimKJeAXQB8AbmBYcIAvVREaIcUQAUASgxKcOFAAsAllAB0+bYlgBeWGctWBiIaLv6FxgL6KxixE5gBunrDunmIS+uTosKie5CisiAQAZokongA0kaoZ5gDmKWmZBfRFxbB+hsbGKmpoCUnC+U3VtY6RgsJinj6BDdgktAwY2XmVXX6+OAEBwAAmiOAQJLAZNPRa5qq9Hv05wnGTbWSw9DQAtgBGiEQdhSXxzsgo5Nf3j91zw0A